futures-util/src/future/join_all.rs

```diff
@@ -50,33 +87,44 @@ where
 /// collecting the results into a destination `Vec<T>` in the same order as they
 /// were provided.
 ///
+/// # Caveats
```

Comment:
I'm not sure whether I would call it `Caveats` or whether I would mention futures-unordered. They seem to have very different use-cases.
The main use-case for `join_all` seems to be to wait for a dynamically created list of futures (opposed to `join!`, which operates on a static list).

Reply:
The performance with a large number of futures is likely to be a good reason to push users towards `futures_unordered` even when `join_all` could work. Maybe "Caveats" is a bit strong though, how do "Alternatives" or "See Also" sound?